tiatoolbox/wsicore/wsireader.py

```diff
@@ -301,13 +303,138 @@ def find_read_rect_params(self, location, size, resolution, units, precision=3):
         ).astype(int)
         level_read_size = np.round(np.array(size) / post_read_scale_factor).astype(int)
         level_location = np.round(np.array(location) / level_downsample).astype(int)
-        return (
+        output = (
             read_level,
             level_location,
             level_read_size,
             post_read_scale_factor,
             baseline_read_size,
         )
+        return output
+
+    def _find_read_params_at_requested_resolution(
+            self, location, size, resolution, units):
+        """Work similar to `find_read_rect_params`.
+
+        This is similar for when location is at requested resolution.
+
+        Args:
+            location (tuple(int)): Location in the requested resolution system.
+            size (tuple(int)): Desired output size in pixels (width,
+                height) tuple and in the requested resolution system.
+            resolution (float): Desired output resolution.
+            units (str): Units of scale, default = "level".
+                Supported units are:
+                - microns per pixel ('mpp')
+                - objective power ('power')
+                - pyramid / resolution level ('level')
+                - pixels per baseline pixel ('baseline')
+            precision (int, optional): Decimal places to use when
+                finding optimal scale. See
+                :func:`find_optimal_level_and_downsample` for more.
+
+        Returns:
+            tuple: Parameters for reading the requested region
+            - :py:obj:`int` - Optimal read level
+            - :py:obj:`tuple` - Scaling to apply after level read to achieve
+              desired output resolution.
+                - :py:obj:`float` - X scale factor
+                - :py:obj:`float` - Y scale factor
+            - :py:obj:`tuple` - Region size in read level coordinates
+                - :py:obj:`int` - Width
+                - :py:obj:`int` - Height
+            - :py:obj:`tuple` - Region location in read level coordinates
+                - :py:obj:`int` - X location
+                - :py:obj:`int` - Y location
+            - :py:obj:`tuple` - Region size in level 0 coordinates
+                - :py:obj:`int` - Width
+                - :py:obj:`int` - Height
+            - :py:obj:`tuple` - Region location level 0 coordinates
+                - :py:obj:`int` - X location
+                - :py:obj:`int` - Y location
+        """
+        (
+            read_level,
+            read_lv_to_requested_fx
```

Comment:
Can we make these variable naming consistent between this function, `_find_read_rect_params` and `_find_optimal_level_and_downsample` docstrings. It's a bit confusing / hard to follow otherwise.